code/modules/client/preference/preferences.dm

```diff
@@ -1148,7 +1148,7 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 				if("name")
 					var/raw_name = input(user, "Choose your character's name:", "Character Preference") as text|null
 					if(!isnull(raw_name)) // Check to ensure that the user entered text (rather than cancel.)
-						var/new_name = reject_bad_name(raw_name)
+						var/new_name = reject_bad_name(raw_name, 2)
```

Comment:
Why not just set this to `var/new_name = reject_bad_name(raw_name, 1)` and make no further changes.